ral-facilities/datagateway-api
# Semantic Versioning

This PR will close #242 

## Description
This PR contains changes for the API's versioning. Once this PR is merged, the API will follow semantic versioning, using a couple of automated tools to manage this. This is explained with the additions in `README.md` but essentially involves using the Angular commit message style to represent what type of release/version bump should be made. I've also added an extra check box for the PR template just as a reminder when you create/review a PR.

Since DataGateway API will be going into production soon and it's essentially ready for this, I have added a `BREAKING CHANGE:` commit which will automatically bump the API to v1.0.0.

This branch also contains commits from #275 to get the CI to pass.

## Testing Instructions
To test, I have forked DataGateway API (https://github.com/MRichards99/datagateway-api), created a [PR equivalent to this](https://github.com/MRichards99/datagateway-api/pull/3) and merged it. You can see that a[ GitHub release of v1.0.0](https://github.com/MRichards99/datagateway-api/releases) has been made on my fork, as intended. Have a look at this and check everything looks all good, or fork DataGateway API yourself and follow the steps I did to see if it works for you.

      deliberate change made to the script to change generated data (which isn't
      actually a problem) or is there an underlying issue with the changes made?

## Agile Board Tracking
Connect to #242
